### PR TITLE
refactor: render based query params

### PR DESCRIPTION
**Pull request by ChristopherPHolder**
closes: https://github.com/push-based/ngx-fast-svg/issues/36

### CODE REVIEW

**Comment by ChristopherPHolder** [comment]
How are we deploying?
We need to update it so we use pre-rendering.

File: packages/ngx-fast-icon-demo/src/app/comparison/angular/angular.component.html [hunk 1]
```diff
@@ -11,13 +11,13 @@ <h4>Layout setting</h4>
...
-  <button (click)="tester.add()">Add</button>
-  <button (click)="tester.remove()">Remove</button>
+  <button (click)="tester.updateList(1)">Add</button>
+  <button (click)="tester.updateList(-1)">Remove</button>
   <button (click)="tester.reload()">Reload page</button>
 </div>
 
-<div class="row icons" *ngIf="tester.showContainer">
-  <ul class="group" *ngFor="let list of tester.lists">
+<div class="row icons" [ngClass]="tester.layout | async">
```
Comment (edbzn) [inline packages/ngx-fast-icon-demo/src/app/comparison/angular/angular.component.html:19]: ngclass has known performance issues

File: packages/ngx-fast-icon-demo/src/app/misc/icons-tester/icon-tester.service.ts [hunk 2]
```diff
@@ -8,67 +11,75 @@ enum ViewportSetting {
...
+          [],
+          {
+            relativeTo: activatedRoute,
+            queryParams: {
+              ls: this._toKebab(ViewportSetting.OnScreen),
+              list: 0
+            },
+            queryParamsHandling: 'merge',
+          });
+      });
```
Comment (edbzn) [inline packages/ngx-fast-icon-demo/src/app/misc/icons-tester/icon-tester.service.ts:75]: No unsubscription is not a good practice.

File: packages/ngx-fast-icon-demo/src/app/comparison/material/material.component.html [hunk 3]
```diff
@@ -11,13 +11,13 @@ <h4>Layout setting</h4>
 </div>
 <h4>Content controls</h4>
 <div class="buttons-wrapper">
-  <button (click)="tester.add()">Add</button>
-  <button (click)="tester.remove()">Remove</button>
+  <button (click)="tester.updateList(1)">Add</button>
+  <button (click)="tester.updateList(-1)">Remove</button>
```
Comment (edbzn) [inline packages/ngx-fast-icon-demo/src/app/comparison/material/material.component.html:15]: What the argument -1 or 1 mean?

File: packages/ngx-fast-icon-demo/src/app/comparison/ant/ant.component.html [hunk 4]
```diff
@@ -2,7 +2,7 @@ <h1>Ant design icon demo</h1>
 <h4>Layout setting</h4>
 <div class="buttons-wrapper">
   <button
-    [ngClass]="{ active: button === tester.setting }"
+    [ngClass]="{ active: button === (tester.setting | async) }"
```
Comment (edbzn) [inline packages/ngx-fast-icon-demo/src/app/comparison/ant/ant.component.html:5]: Same, ngclass should be avoided.

File: packages/ngx-fast-icon-demo/src/app/comparison/ionic/ionic.component.html [hunk 5]
```diff
@@ -11,13 +11,13 @@ <h4>Layout setting</h4>
...
-  <button (click)="tester.add()">Add</button>
-  <button (click)="tester.remove()">Remove</button>
+  <button (click)="tester.updateList(1)">Add</button>
+  <button (click)="tester.updateList(-1)">Remove</button>
   <button (click)="tester.reload()">Reload page</button>
 </div>
 
-<div class="row icons" *ngIf="tester.showContainer">
-  <ul class="group" *ngFor="let list of tester.lists">
+<div class="row icons" [ngClass]="tester.layout | async">
```
Comment (edbzn) [inline packages/ngx-fast-icon-demo/src/app/comparison/ionic/ionic.component.html:19]: The usage of async will impact performance.

**Comment by edbzn** [comment]
Left a couple of comments @ChristopherPHolder.

**Pull request merged**